google/cloud/storage/blob.py

```diff
@@ -3873,23 +3887,38 @@ def open(
                 raise ValueError(
                     "encoding, errors and newline arguments are for text mode only"
                 )
+            if ignore_flush:
+                raise ValueError(
+                    "ignore_flush argument is for non-text write mode only"
+                )
```

Comment:
Can we move this check out to a single point, e.g.:

```python

        if ignore_flush and mode in ("r", "rt", "rb", "w", "wt"):
            raise ValueError(
                "ignore_flush argument is for non-text write mode only"
            )
```

Reply:
This would only deduplicate two of the three ValueErrors, because the third has a different behavior (errors on false) and a different error message. To keep consistent organization I'd prefer to keep the current pattern, even if it's more verbose.

Reply:
OK -- I'd missed that the error case was reversed for `w` / `wt`.

One might argue that pulling the validation out from the "construct the IO helper" would make that clearer, but I will defer here.